modules/bioinspired/src/retina.cpp

```diff
@@ -295,8 +296,21 @@ class RetinaImpl : public Retina
 };
 
 // smart pointers allocation :
-Ptr<Retina> createRetina(Size inputSize){ return makePtr<RetinaImpl>(inputSize); }
-Ptr<Retina> createRetina(Size inputSize, const bool colorMode, int colorSamplingMethod, const bool useRetinaLogSampling, const float reductionFactor, const float samplingStrenght){
+Ptr<Retina> createRetina(Size inputSize)
+{
+#ifdef HAVE_OPENCL
+    if (cv::ocl::useOpenCL() && (inputSize.width % 4 == 0))
+        return makePtr<cv::bioinspired::ocl::RetinaOCLImpl>(inputSize);
+#endif
+    return makePtr<RetinaImpl>(inputSize);
```

Comment:
I believe there are should be one implementation that is able to handle OCL and CPU code paths (via `OCL_RUN` hooks).
But there is main problem with data storage: Mat <-> UMat conversion is not cheap now for CPU data storage (there is forced OpenCL buffer creation).